CONTRIBUTING.md

```diff
@@ -33,6 +33,47 @@ sounds interesting](https://github.com/metal3-io/cluster-api-provider-metal3/iss
 Alternatively, read some of the docs on other controllers and try to write your
 own, file and fix any/all issues that come up, including gaps in documentation!
 
+## Versioning
+
+### Codebase and Go Modules
+
+> :warning: The project does not follow Go Modules guidelines for compatibility requirements for 1.x semver releases.
+Cluster API Metal3 follows Cluster API release cadence abd versioning which follows upstream Kubernetes semantic versioning. With the v1 release of our codebase, we guarantee the following:
+
+- A (*minor*) release CAN include:
+  - Introduction of new API versions, or new Kinds.
+  - Compatible API changes like field additions, deprecation notices, etc.
+  - Breaking API changes for deprecated APIs, fields, or code.
+  - Features, promotion or removal of feature gates.
+  - And more!
+
+- A (*patch*) release SHOULD only include backwards compatible set of bugfixes.
+
+These guarantees extend to all code exposed in our Go Module, including
+*types from dependencies in public APIs*.
+Types and functions not in public APIs are not considered part of the guarantee.
+The test module, and experiments do not provide any backward compatible guarantees.
+
+#### Backporting
+
+We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
+backport MUST not be breaking for either API or behavioral changes.
+We generally do not accept PRs against older release branches.
```

Comment:
Q: I remember this being discussed recently upstream, however not sure if we should really follow it as a provider? https://github.com/kubernetes-sigs/cluster-api/pull/7167

Reply:
I would prefer to keep it as it is now.